### PR TITLE
RDCC-1584: Increasing Mutation Coverage to meet threshold

### PR DESCRIPTION
**Pull request by AdilOoze**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/RDCC-1586

### Change description ###

Adding unit tests to increase mutation coverage & small refactoring of code 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java [hunk 1]
```diff
@@ -195,14 +228,45 @@ public void deleteRoles_InternalServerError() throws Exception {
...
+        verify(idamFeignClientMock, times(1)).deleteUserRole(any(String.class), any(String.class));
+        verify(responseMock, times(1)).close();
+    }
+
+    @Test
+    public void deleteRoles_FeignException() {
+        RoleName roleName1 = new RoleName("pui-case-manager");
+        Set<RoleName> roles = new HashSet<>();
+        roles.add(roleName1);
+        updateUserProfileData.setRolesDelete(roles);
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java:250]: prefix test name with test

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/util/UserProfileMapperTest.java [hunk 2]
```diff
@@ -73,10 +80,15 @@ public void test_mapIdamUpdateStatusRequestWhenStatusIsSuspended() {
...
-        assertThat(userProfile.getLastName()).isEqualTo(userProfileCreationData.getLastName());
+        UserProfile userProfileMock = mock(UserProfile.class);
+        when(userProfileMock.getFirstName()).thenReturn(userProfileCreationData.getFirstName());
+        when(userProfileMock.getLastName()).thenReturn(userProfileCreationData.getLastName());
+
+        UserProfileMapper.mapUpdatableFieldsForReInvite(userProfileCreationData, userProfileMock);
+        
+        assertThat(userProfileMock.getFirstName()).isEqualTo(userProfileCreationData.getFirstName());
+        assertThat(userProfileMock.getLastName()).isEqualTo(userProfileCreationData.getLastName());
+        verify(userProfileMock, times(1)).setLastUpdated(any(LocalDateTime.class));
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/util/UserProfileMapperTest.java:91]: if possible add one more verify for when condition

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java [hunk 3]
```diff
@@ -165,20 +167,51 @@ public void addRoles_InternalServerError() throws Exception {
...
+        assertThat(response.getRoleAdditionResponse().getIdamStatusCode()).isEqualTo("500");
+
+        verify(userProfileRepositoryMock, times(1)).findByIdamId(any(String.class));
+        verify(idamFeignClientMock, times(1)).addUserRoles(any(), any(String.class));
+        verify(responseMock, times(1)).close();
+    }
+
+    @Test
+    public void addRoles_FeignException() {
+        RoleName roleName1 = new RoleName("pui-case-manager");
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java:186]: prefix test for method

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java [hunk 4]
```diff
@@ -165,20 +167,51 @@ public void addRoles_InternalServerError() throws Exception {
...
-        verify(idamFeignClientMock, times(1)).addUserRoles(any(),any(String.class));
+        verify(idamFeignClientMock, times(1)).addUserRoles(any(), any(String.class));
+        verify(auditServiceMock, times(1)).persistAudit(any(HttpStatus.class), any(UserProfile.class), any(ResponseSource.class));
     }
 
     @Test
     public void deleteRoles_InternalServerError() throws Exception {
         RoleName roleName1 = new RoleName("pui-case-manager");
-        // RoleName roleName2 = new RoleName("pui-case-organisation");
         Set<RoleName> roles = new HashSet<>();
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java:215]: prefix test for method

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java [hunk 5]
```diff
@@ -194,13 +243,45 @@ public void should_retrieve_user_multiple_profiles_without_roles_when_idam_fails
...
+
+        verify(auditRepository, Mockito.times(1)).save(any(Audit.class));
+        verify(upMock, times(1)).setRoles(any(IdamRolesInfo.class));
+        verify(upMock, times(1)).setErrorMessage(any(String.class));
+        verify(upMock, times(1)).setErrorStatusCode(any(String.class));
+    }
+
     @Test
     public void should_throw_404_single_user_profile_without_roles_when_idam_fails() {
         idamRolesInfo = new IdamRolesInfo(entity, HttpStatus.NOT_FOUND);
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java:287]: prefix test for method

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java [hunk 6]
```diff
@@ -194,13 +243,45 @@ public void should_retrieve_user_multiple_profiles_without_roles_when_idam_fails
...
-        assertThat(profile.getRoles().size()).isEqualTo(0);
+        assertThat(profile.getRoles().size()).isZero();
         assertThat(profile.getErrorMessage()).isNotEmpty();
         assertThat(profile.getErrorStatusCode()).isEqualTo("404");
 
         Mockito.verify(auditRepository, Mockito.times(1)).save(any(Audit.class));
     }
 
+    @Test
+    public void should_retrieve_user_multiple_profiles_with_roles_when_idam_success() {
```
Comment (KOTLAPRA23) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java:254]: prefix test for method

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java (hunk 4 above)
Comment (KOTLAPRA23) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java:215]: remove commented code

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java (hunk 6 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java:254]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java (hunk 5 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java:287]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java (hunk 4 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java:214]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java (hunk 3 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java:186]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/util/UserProfileMapperTest.java (hunk 2 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/util/UserProfileMapperTest.java:91]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java (hunk 1 above)
Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileUpdatorTest.java:249]: This has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/04d04c3144cda8c41c2c8b875f3a62b0fa855501

File: src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java [hunk 7]
```diff
@@ -167,6 +184,38 @@ public void should_retrieve_Multiple_Profiles() {
...
+        assertThat(getUserProfile1.getLastName()).isEqualTo(up1.getLastName());
+        assertThat(getUserProfile1.getRoles()).isEqualTo(up1.getRoles());
+        assertThat(getUserProfile1.getErrorMessage()).isEqualTo(up1.getErrorMessage());
+        assertThat(getUserProfile1.getErrorStatusCode()).isEqualTo(up1.getErrorStatusCode());
+        verify(querySupplier, times(1)).getProfilesByIds(identifier, true);
+    }
+
     @Test
     public void should_throw_404_when_no_profiles_found_in_db() {
         UserProfileIdentifier identifier = mock(UserProfileIdentifier.class);
```
Comment (shashank-rastogi) [inline src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java:221]: all tests in the class do not have test as prefix.. if this is a standard practice followed then it should be updated for all unit tests

Comment (AdilOoze) [reply src/test/java/uk/gov/hmcts/reform/userprofileapi/service/impl/UserProfileRetrieverTest.java:221]: this has been addressed in the following commit: https://github.com/hmcts/rd-user-profile-api/pull/385/commits/8d2cf5fd8500a9e186c888746072467e7fcf8a3e

File: src/functionalTest/resources/application-functional.yaml [hunk 8]
```diff
@@ -10,6 +10,8 @@ test.user.password: ${TEST_USER_PASSWORD:Hmcts123}
 s2s.auth.secret: ${S2S_SECRET:OLMEJQA7UHU6EZKG}
 s2s.auth.url: ${S2S_URL_TO_TEST:http://rpe-service-auth-provider-aat.service.core-compute-aat.internal}
 
+s2s-name: rd_user_profile_api
+
```
Comment (shashank-rastogi) [inline src/functionalTest/resources/application-functional.yaml:14]: is this needed?

Comment (AdilOoze) [reply src/functionalTest/resources/application-functional.yaml:14]: All Functional Test cases were failing with 401 when attempting to get S2S Token as it was picking rd-professional-api name and secret - so including the name and secret key resolved the issue

**Pull request merged**